services/traction/api/db/migrations/versions/v1_error_status_detail_8ac2e131039d.py

```diff
@@ -0,0 +1,223 @@
+# flake8: noqa
+"""v1-error_status_detail
+
+Revision ID: 8ac2e131039d
+Revises: b01986f67aa3
+Create Date: 2022-06-09 13:34:45.813263
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import sqlmodel
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "8ac2e131039d"
+down_revision = "b01986f67aa3"
+branch_labels = None
+depends_on = None
+
+
+create_error_status_func = """CREATE OR REPLACE FUNCTION error_status_detail_func() RETURNS trigger AS $body$
+    BEGIN
+        IF NEW.status <> 'Error' THEN
+            NEW.error_status_detail = NULL;
+        END IF;
+        RETURN NEW;
+    END;
+    $body$ LANGUAGE plpgsql
+"""
```

Comment:
why do we need these as triggers? this should be enforced at the code level if this is critical.

Reply:
it's just easier to get all the cases done this way than enforce it in code. there is no single update function.

Reply:
See comment in basemodel, i think there is a one line global fix.